File: app/src/main/java/com/asmat/rolando/popularmovies/activities/SearchResultsActivity.java
package com.asmat.rolando.popularmovies.activities;

import android.app.SearchManager;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.SearchView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuInflater;
import android.widget.Toast;

import com.asmat.rolando.popularmovies.R;
import com.asmat.rolando.popularmovies.fragments.SearchResultsFragment;


public class SearchResultsActivity extends AppCompatActivity {
    SearchResultsFragment mResultsGrid;
    SearchView mSearchview;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search_results);
        setupToolbar();
        setupFragment();
        handleIntent(getIntent());
    }

    @Override
    protected void onNewIntent(Intent intent) {
        setIntent(intent);
        handleIntent(intent);
    }

    // Enable Back arrow
    private void setupToolbar() {
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        ActionBar bar = getSupportActionBar();
        if (bar != null) {
            bar.setDisplayHomeAsUpEnabled(true);
        }
    }

    private void setupFragment() {
        mResultsGrid = (SearchResultsFragment) getSupportFragmentManager().findFragmentById(R.id.results_grid);
    }

    // Handle the search query
    private void handleIntent(Intent intent) {
        if (Intent.ACTION_SEARCH.equals(intent.getAction())) {
            String query = intent.getStringExtra(SearchManager.QUERY);
            mSearchview.setQuery(query, false);
            mResultsGrid.setSearchQuery(query);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the options menu from XML
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_search_results, menu);
        // Get the SearchView and set the searchable configuration
        SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
        mSearchview = (SearchView) menu.findItem(R.id.search).getActionView();
        // Assumes current activity is the searchable activity
        mSearchview.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
        mSearchview.setIconified(false);
        mSearchview.setIconifiedByDefault(true);
        mSearchview.setSubmitButtonEnabled(true);
        return true;
    }
}

File: app/src/main/java/com/asmat/rolando/popularmovies/ui/search/SearchResultsActivity.kt
package com.asmat.rolando.popularmovies.ui.search

import android.app.SearchManager
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView
import androidx.appcompat.widget.Toolbar
import android.view.Menu

import com.asmat.rolando.popularmovies.R
import com.asmat.rolando.popularmovies.ui.fragments.SearchResultsFragment

class SearchResultsActivity : AppCompatActivity() {
    lateinit var resultsGrid: SearchResultsFragment
    lateinit var searchview: SearchView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_search_results)
        setupToolbar()
        setupFragment()
    }

    override fun onNewIntent(intent: Intent) {
        setIntent(intent)
        handleIntent(intent)
    }

    // Enable Back arrow
    private fun setupToolbar() {
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        val bar = supportActionBar
        bar?.setDisplayHomeAsUpEnabled(true)
    }

    private fun setupFragment() {
        resultsGrid = supportFragmentManager.findFragmentById(R.id.results_grid) as SearchResultsFragment
    }

    // Handle the search query
    private fun handleIntent(intent: Intent) {
        if (Intent.ACTION_SEARCH == intent.action) {
            val query = intent.getStringExtra(SearchManager.QUERY)
            searchview.setQuery(query, false)
            searchview.clearFocus()
            resultsGrid.setSearchQuery(query)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the options menu from XML
        val inflater = menuInflater
        inflater.inflate(R.menu.menu_search_results, menu)
        // Get the SearchView and set the searchable configuration
        val searchManager = getSystemService(Context.SEARCH_SERVICE) as SearchManager
        searchview = menu.findItem(R.id.search).actionView as SearchView
        // Assumes current activity is the searchable activity
        searchview.setSearchableInfo(searchManager.getSearchableInfo(componentName))
        searchview.isIconified = false
        searchview.setIconifiedByDefault(true)
        searchview.isSubmitButtonEnabled = true
        handleIntent(intent)
        return true
    }
}